lib/sie/parser/tokenizer.rb

```diff
@@ -16,7 +16,7 @@ def tokenize
           move_to_next_character
           break unless current_character.value
 
-          if consume?
+          if consume? && !current_character.end_of_array?
```

Comment:
Is it possible to change consume to false when adding an end_of_array token below? Or does that break things?